src/detect.c

```diff
@@ -292,6 +292,70 @@ char *DetectLoadCompleteSigPath(const DetectEngineCtx *de_ctx, char *sig_file)
     return path;
 }
 
+static SigString *SigStringAlloc(void)
```

Comment:
Thinking we should maybe put this in a new/different file. Trying to reduce the size of detect.c and this seems more like a util than a core detect feature.